src/main/java/data/SpendingAnalytics.java

```diff
@@ -17,27 +17,63 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.stream.Collectors;
 
 /** Class that computes and stores user's spending analytics. */
 public class SpendingAnalytics {
   private final HashMap<String, Double> storeAnalytics;
+  private final HashMap<String, Double> categoryAnalytics;
 
   public SpendingAnalytics(ImmutableSet<Entity> allReceipts) {
     storeAnalytics = new HashMap<>();
+    categoryAnalytics = new HashMap<>();
 
     for (Entity receipt : allReceipts) {
-      String store = (String) receipt.getProperty("store");
-      Double price = (Double) receipt.getProperty("price");
+      updateStoreAnalytics(receipt, storeAnalytics);
 
-      // Don't add to hashmap if either store or price are invalid.
-      if (!Strings.isNullOrEmpty(store) && price != null) {
-        if (storeAnalytics.containsKey(store)) {
-          storeAnalytics.put(store, storeAnalytics.get(store) + price);
+      updateCategoryAnalytics(receipt, categoryAnalytics);
```

Comment:
Are `storeAnalytics` and `categoryAnalytics` necessary to include as arguments? They can be accessed as instance variables.